src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -855,3 +855,165 @@ end
+  [hP : P.is_prime] :
+  ∏ i in s, f i ≤ P ↔ ∃ i ∈ s, f i ≤ P :=
+begin
+  by_cases hP0 : P = ⊥,
+  { simp only [hP0, le_bot_iff],
+    rw [← ideal.zero_eq_bot, finset.prod_eq_zero_iff] },
+  simp only [← ideal.dvd_iff_le],
+  exact ((ideal.prime_iff_is_prime hP0).mpr hP).dvd_finset_prod_iff _
+end
+
+/-- The intersection of distinct prime powers in a Dedekind domain is the product of these
+prime powers. -/
+lemma is_dedekind_domain.inf_prime_pow_eq_prod {ι : Type*}
+  (s : finset ι) (f : ι → ideal R) (e : ι → ℕ)
+  (prime : ∀ i, prime (f i)) (coprime : ∀ i j, i ≠ j → f i ≠ f j) :
+  s.inf (λ i, f i ^ e i) = ∏ i in s, f i ^ e i :=
+begin
+  letI := classical.dec_eq ι,
+  refine s.induction _ _,
+  { simp },
+  intros a s ha ih,
+  rw [finset.inf_insert, finset.prod_insert ha, ih],
+  refine le_antisymm (ideal.le_mul_of_no_prime_factors _ inf_le_left inf_le_right) ideal.mul_le_inf,
+  intros P hPa hPs hPp,
+  haveI := hPp,
+  obtain ⟨b, hb, hPb⟩ := ideal.prod_le_prime.mp hPs,
+  haveI := ideal.is_prime_of_prime (prime a), haveI := ideal.is_prime_of_prime (prime b),
+  refine coprime a b _
+    (((is_dedekind_domain.dimension_le_one.prime_le_prime_iff_eq _).mp
+        (ideal.le_of_pow_le_prime hPa)).trans
+      ((is_dedekind_domain.dimension_le_one.prime_le_prime_iff_eq _).mp
+        (ideal.le_of_pow_le_prime hPb)).symm),
+  { unfreezingI { rintro rfl }, contradiction },
+  { exact (prime a).ne_zero },
+  { exact (prime b).ne_zero },
+end
+
+/-- **Chinese remainder theorem** for a Dedekind domain: if the ideal `I` factors as
+`∏ i, P i ^ e i`, then `R ⧸ I` factors as `Π i, R ⧸ (P i ^ e i)`. -/
+noncomputable def is_dedekind_domain.quotient_equiv_pi_of_prod_eq {ι : Type*} [fintype ι]
+  (I : ideal R) (P : ι → ideal R) (e : ι → ℕ)
+  (prime : ∀ i, prime (P i)) (coprime : ∀ i j, i ≠ j → P i ≠ P j) (prod_eq : (∏ i, P i ^ e i) = I) :
+  R ⧸ I ≃+* Π i, R ⧸ (P i ^ e i) :=
+(ideal.quot_equiv_of_eq (by { simp only [← prod_eq, finset.inf_eq_infi,
+  ← is_dedekind_domain.inf_prime_pow_eq_prod _ _ _ prime coprime, finset.mem_univ, cinfi_pos] }))
+    .trans $
+ideal.quotient_inf_ring_equiv_pi_quotient _ (λ i j hij, ideal.coprime_of_no_prime_ge (begin
+  intros P hPi hPj hPp,
+  haveI := hPp,
+  haveI := ideal.is_prime_of_prime (prime i), haveI := ideal.is_prime_of_prime (prime j),
+  exact coprime i j hij
+    (((is_dedekind_domain.dimension_le_one.prime_le_prime_iff_eq (prime i).ne_zero).mp
+      (ideal.le_of_pow_le_prime hPi)).trans
+    ((is_dedekind_domain.dimension_le_one.prime_le_prime_iff_eq (prime j).ne_zero).mp
+     (ideal.le_of_pow_le_prime hPj)).symm)
+end))
+
+open_locale classical
+
+/-- **Chinese remainder theorem** for a Dedekind domain: `R ⧸ I` factors as `Π i, R ⧸ (P i ^ e i)`,
+where `P i` ranges over the prime factors of `I` and `e i` over the multiplicities. -/
+noncomputable def is_dedekind_domain.quotient_equiv_pi_factors {I : ideal R} (hI : I ≠ ⊥) :
```

Comment:
Is this really false for `I = ⊥`? I am maybe confused, but the product would be over the empty set, so it seems to work.

Reply:
The quotient by ⊥ is the whole ring `R` and the product over the empty set would be the trivial ring, right? So I don't think this holds for `I = ⊥`.

Reply:
You're right, sorry.